File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int: #
        left = 0
        right = len(nums)-1
        center = math.floor((right-left)/2)

        while left != center and right != center:
            if nums[center] > nums[right]:
                left = center
                center = left + math.floor((right-left) /2)
            else:
                right = center
                center = left + math.floor((right-left) /2)
            
        return min(nums[left], nums[right])

File: Data Structures & Algorithms/is-palindrome/submission-0.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string s_alnum;
        for(auto c : s){
            if(isalnum(c)){
                s_alnum += tolower(c);
            }
        }
        s = s_alnum;
        int left = 0;
        int right = s.size() - 1;

        while(left < right){
            if(s[left] != s[right]){
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
};
